File: backend/src/main/java/com/golf/model/Course.java
package com.golf.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "courses")
@Data
public class Course {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(nullable = false)
    private String name;

    @NotBlank
    private String location;

    @Min(1) @Max(18)
    @Column(nullable = false)
    private Integer numberOfHoles = 18;

    /** Course rating from authoritative golf association */
    private Double courseRating;

    /** Slope rating (55-155) */
    @Min(55) @Max(155)
    private Integer slopeRating;

    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("holeNumber ASC")
    private List<Hole> holes = new ArrayList<>();
}

File: mobile-app/lib/models/course.dart
class Course {
  final int id;
  final String name;
  final String location;
  final int numberOfHoles;
  final double? courseRating;
  final int? slopeRating;

  const Course({
    required this.id,
    required this.name,
    required this.location,
    required this.numberOfHoles,
    this.courseRating,
    this.slopeRating,
  });

  factory Course.fromJson(Map<String, dynamic> json) => Course(
        id: json['id'] as int,
        name: json['name'] as String,
        location: json['location'] as String,
        numberOfHoles: json['numberOfHoles'] as int,
        courseRating: (json['courseRating'] as num?)?.toDouble(),
        slopeRating: json['slopeRating'] as int?,
      );

  Map<String, dynamic> toJson() => {
        'name': name,
        'location': location,
        'numberOfHoles': numberOfHoles,
        if (courseRating != null) 'courseRating': courseRating,
        if (slopeRating != null) 'slopeRating': slopeRating,
      };
}
